content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+```
+
+You can now use this Function in your Composition. The below example will create
+an `RDSInstance` using P&T Composition, then run the Function to annotate it
+with a quote.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example
+spec:
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: XPostgreSQLInstance
+  resources:
+    - name: rds-instance
+      base:
+        apiVersion: rds.aws.upbound.io/v1beta1
+        kind: Instance
+        spec:
+          forProvider:
+            dbName: exmaple
+            instanceClass: db.t3.micro
+            region: us-west-2
+            skipFinalSnapshot: true
+            username: exampleuser
+            engine: postgres
+            engineVersion: "12"
+      patches:
+        - fromFieldPath: spec.parameters.storageGB
+          toFieldPath: spec.forProvider.allocatedStorage
+      connectionDetails:
+        - type: FromFieldPath
+          name: username
+          fromFieldPath: spec.forProvider.username
+        - type: FromConnectionSecretKey
+          name: password
+          fromConnectionSecretKey: attribute.password
+  functions:
+  - name: quotable
+    type: Container
+    container:
+      image: xpkg.upbound.io/example-org/xfn-quotable-simple:v0.1.0
+      network:
+        policy: Runner
+```
+
+### Tips for New Functions
+
+Here are some things to keep in mind when building a Composition Function:
+
+* Your Function may be running as part of a pipeline. This means it's very
+  important that your Function pass through any desired state that it is
+  unconcerned with. If your Function is passed a desired composed resource and
+  does not return that composed resource in its output, it will be deleted.
+  Crossplane considers the desired state of the XR and any composed resources to
+  be whatever `FunctionIO` is returned by the last Function in the pipeline.
+* Crossplane won't set a `metadata.name` for your desired resources resources.
+  It's a good practice to match P&T Composition's behavior by setting
+  `metadata.generateName: "name-of-the-xr-"` for any new desired resources.
+* Don't blindly add new entries to the desired resources array every time your
+  function is invoked. Remember to check whether your desired resource is
+  already in the `observed` and/or `desired` objects. You may need to update it
+  rather than create it.
+* Don't bypass providers. Composition Functions are designed to tell Crossplane
+  how to orchestrate managed resources - not to directly orchestrate external
+  systems.
+* Include your function name and version in any results you return to aid in
+  debugging.
+* Write tests for your function! Pass it a `FunctionIO` on stdin in and ensure
+  it returns the expected `FunctionIO` on stdout.
+* Keep your Functions fast and lightweight. Remember that Crossplane will run
+  them approximately once every 30-60 seconds.
```

Comment:
```suggestion
  them every 30 to 60 seconds.
```
Don't use "approximately", be precise, even with a range.

Reply:
Why? The quoted numbers are actually approximate. It may be less than 30 seconds in some cases.

Reply:
It's a [weasel word](https://en.wikipedia.org/wiki/Weasel_word) (https://github.com/btford/write-good). 
Since we're specifying a range an approximation is implied.